### PR TITLE
[CacheRefresh] Class refactor

### PR DESCRIPTION
**Pull request by shule727**

### CODE REVIEW

**Review by lskupnjak — APPROVED** [review]

File: CacheRefresh.php [hunk 1]
```diff
@@ -2,53 +2,39 @@
...
+    private ExpressionLanguage $expressionLanguage;
 
     public function __construct(LoaderInterface $mappingsLoader, PropertyAccessorInterface $propertyAccessor)
     {
         $this->mappingsLoader = $mappingsLoader;
         $this->propertyAccessor = $propertyAccessor;
 
-        if (!class_exists('Symfony\Component\ExpressionLanguage\ExpressionLanguage')) {
-            throw new \RuntimeException('Unable to use expressions as the Symfony ExpressionLanguage component is not installed.');
+        if (!class_exists(ExpressionLanguage::class)) {
```
Comment (spideyfusion) [inline CacheRefresh.php:30]: @shule727 Slobodno makni kompletno ovu provjeru pošto je expression language komponenta postavljena kao dependency kroz Composer.

**Review by spideyfusion — COMMENTED** [review]
Please dodaj i testove za ovu klasu da podignemo taj test coverage.

**Review by plantas — APPROVED** [review]

**Review by dominikgalicc — APPROVED** [review]

File: CacheRefresh.php [hunk 2]
```diff
@@ -2,53 +2,34 @@
...
 
+use Exception;
+use RuntimeException;
 use SofaScore\Purgatory\Mapping\Loader\LoaderInterface;
 use SofaScore\Purgatory\Mapping\MappingCollection;
-use SofaScore\Purgatory\Mapping\MappingValue;
 use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
 use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
 class CacheRefresh
```
Comment (jstuhli) [inline CacheRefresh.php:12]: Might as well rename the class when we're refactoring it. The class isn't actually doing anything related to Cache Refresh :).
Maybe we could just call it Purgatory as a main entry point?

File: CacheRefresh.php [hunk 3]
```diff
@@ -67,9 +48,8 @@ public function getMappings()
      * @param mixed $object            Object that was changed
      * @param array $changedProperties List of property paths (ex. ['status.description', 'userCount', ...])
      *
-     * @return array
      */
-    public function getUrlsToRefresh($object, $changedProperties)
+    public function getUrlsToRefresh($object, array $changedProperties): array
```
Comment (jstuhli) [inline CacheRefresh.php:52]: Might be a good time to also rename this method. The urls are purged not refreshed, so at the very least this should be getUrlsToPurge.

File: CacheRefresh.php [hunk 4]
```diff
@@ -195,7 +172,7 @@ public function processMappingValues($object, array $mappingValues, array &$urls
 
             // resolve priority
             $priority = $mappingValue->getPriority();
-            $priority = null === $priority ? '@'.self::PRIORITY_DEFAULT : $priority;
+            $priority = $priority ?? ('@' . self::PRIORITY_DEFAULT);
```
Comment (jstuhli) [inline CacheRefresh.php:175]: priority is removed, make sure to rebase propertly

**Review by jstuhli — APPROVED** [review]

File: CacheRefresh.php (hunk 2 above)
Comment (shule727) [reply CacheRefresh.php]: CacheRefresh is now renamed to Purgatory.

File: CacheRefresh.php (hunk 3 above)
Comment (shule727) [reply CacheRefresh.php]: Method is now renamed to `getUrlsToPurge`.

**Review by zlatkoverk — APPROVED** [review]

**Pull request merged**